src/io/flutter/perf/PerfService.java

```diff
@@ -46,22 +56,33 @@ public void connectionClosed() {
       }
     });
 
-    vmService.streamListen(VmService.GC_STREAM_ID, VmServiceConsumers.EMPTY_SUCCESS_CONSUMER);
-    vmService.streamListen(VmService.EXTENSION_STREAM_ID, VmServiceConsumers.EMPTY_SUCCESS_CONSUMER);
     vmService.streamListen(VmService.ISOLATE_STREAM_ID, VmServiceConsumers.EMPTY_SUCCESS_CONSUMER);
+    vmService.streamListen(VmService.EXTENSION_STREAM_ID, VmServiceConsumers.EMPTY_SUCCESS_CONSUMER);
+    vmService.streamListen(VmService.GC_STREAM_ID, VmServiceConsumers.EMPTY_SUCCESS_CONSUMER);
 
     // Populate the service extensions info.
     vmService.getVM(new VMConsumer() {
       @Override
       public void received(VM vm) {
-        for (IsolateRef ref : vm.getIsolates()) {
-          vmService.getIsolate(ref.getId(), new GetIsolateConsumer() {
+        for (final IsolateRef isolateRef : vm.getIsolates()) {
+          vmService.getIsolate(isolateRef.getId(), new GetIsolateConsumer() {
             @Override
             public void onError(RPCError error) {
             }
 
             @Override
             public void received(Isolate isolate) {
+              // Populate flutter isolate info.
+              if (flutterIsolateRef == null) {
+                for (String extensionName : isolate.getExtensionRPCs()) {
+                  if (extensionName.startsWith("ext.flutter.")) {
+                    flutterIsolateRef = isolateRef;
```

Comment:
maybe assert that there is only one flutter isolate 
and keep the first one?
It is theoretically possible someone could add ex.flutter extensions on a background isolate.

Reply:
Currently, this would happen if there was more than one flutter view running in the app. This is possible, but not a use case that we hit currently. I'd prefer to take the first flutter view and ignore the others instead of asserting (an assert will manifest as a crash to the user).

Perhaps just adding a TODO: to note that the best effort handling of multiple flutter view can be improved in the future?